src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -328,8 +436,8 @@ internal static ulong[] InternalGetValues(RuntimeType enumType)
         public static bool IsDefined<TEnum>(TEnum value) where TEnum : struct, Enum
         {
             RuntimeType enumType = (RuntimeType)typeof(TEnum);
-            ulong[] ulValues = Enum.InternalGetValues(enumType);
-            ulong ulValue = Enum.ToUInt64(value);
+            ulong[] ulValues = InternalGetValues(enumType);
```

Comment:
It may work better to switch the cached values type to be the underlying enum type. It would eliminate the conversions to UInt64 (we would have to convert or switch over the underlying type instead), and it would also make the cache 2x smaller for the typical case.

`TEnum[] GetValues<TEnum>()` above would then turn into:

```
    Array values = InternalGetValues(type);
    var results = new TEnum[values.Length];
    Array.Copy(values, results, results.Length); // This can call Buffer.MemMove directly if we really cared about squeezing last cycle out of it
    return results;
```

Reply:
Maybe.  That makes some other APIs more challenging, like InternalFlagsFormat.  I can open an issue to look at doing so, but I'll skip it in this PR.  And since it's complicating things, I'll just revert my GetValues changes.